### PR TITLE
Renames XpringClient to XRPClient

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
XRPClient is a more appropriate name for the client class in Xpring SDKs.  This PR begins the process of updating this name while maintaining the current implementation of XpringClient for gradual deprecation. 

### Context of Change
This PR updates all "XpringClient" related names in the SDK to "XRPClient", such as "DefaultXpringClient" --> "DefaultXRPClient", as well as local variables such as "xpringClient" --> "xrpClient".  It also introduces a new top-level client class "XRPClient" while maintaining the original "XpringClient" class for eventual deprecation.  The "XpringClient" class now wraps an instance of "DefaultXRPClient".

Analogues in other SDKs:
https://github.com/xpring-eng/XpringKit/pull/134
xpring-eng/xpring4j#113

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)

## Before / After
Before: Everything named as "XpringClient"
After: top level "XpringClient" still exists, but everything else is now "XRPClient", including identical top-level class by this name.

## Test Plan
Circle CI

Future tasks include updating names correspondingly in the other SDKs, and eventually deprecating the XpringClient class.

### CODE REVIEW

File: .mocharc.js [hunk 1]
```diff
@@ -1,23 +0,0 @@
-'use strict';
```
Comment (keefertaylor) [inline .mocharc.js]: please revert this file

File: .vscode/settings.json [hunk 2]
```diff
@@ -18,5 +18,11 @@
     },
     "files.insertFinalNewline": true,
     "files.trimFinalNewlines": true,
-    "files.trimTrailingWhitespace": true
+    "files.trimTrailingWhitespace": true,
```
Comment (keefertaylor) [inline .vscode/settings.json]: I think that Tyler and I fixed this this morning if you merge master. LMK if that doesn't work for you.

File: .mocharc.js (hunk 1 above)
Comment (amiecorso) [reply .mocharc.js]: will do - any idea why that happened?  this wasn't a line I manually deleted

File: .vscode/settings.json (hunk 2 above)
Comment (amiecorso) [reply .vscode/settings.json]: I tried merging master and got an 'Already up to date'... i'm not sure what to do about that

Comment (amiecorso) [reply .vscode/settings.json]: This is taken care of now

**Comment by amiecorso** [comment]
Aware that this is broken, need help figuring out why integration test is failing.  When truly ready for review, will add another reviewer.

**Comment by keefertaylor** [comment]
We should also rename `grpc-network-client.ts` to `grpc-xrp-network-client.ts` and associated classes in this PR.

File: .mocharc.js (hunk 1 above)
Comment (amiecorso) [reply .mocharc.js]: reverted

File: README-ja.md [hunk 3]
```diff
@@ -116,42 +116,42 @@ const signature = wallet.sign(message);
...
-「XpringClient」はXRPレジャーへのゲートウェイです。 「XpringClient」は、リモートアダプターのURLである単一のパラメーターで初期化されます（上記の「サーバー側コンポーネント」セクションを参照）。
+「XRPClient」はXRPレジャーへのゲートウェイです。 「XRPClient」は、リモートアダプターのURLである単一のパラメーターで初期化されます（上記の「サーバー側コンポーネント」セクションを参照）。
 
 ```javascript
-const { XpringClient } = require("xpring-js");
+const { XRPClient } = require("xpring-js");
 
 const remoteURL = "grpc.xpring.tech:80";
-const xpringClient = XpringClient.xpringClientWithEndpoint(remoteURL);
+const xrpClient = XRPClient.xrpClientWithEndpoint(remoteURL);
```
Comment (keefertaylor) [inline README-ja.md]: We should probably update these code snippets to be on par with the english version. Can you either file an asana talk or update them?

File: test/mocha.opts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+--require ./node_modules/ts-node/register
```
Comment (keefertaylor) [inline test/mocha.opts]: You don't need to add this.

**Comment by hbergren** [comment]
@amiecorso , can you resolve conflicts before I review?

File: test/mocha.opts (hunk 4 above)
Comment (amiecorso) [reply test/mocha.opts]: Interesting, I didn't add this.  Any ideas why this was auto-generated?

Comment (amiecorso) [reply test/mocha.opts]: (removed anyway)

File: README-ja.md (hunk 3 above)
Comment (amiecorso) [reply README-ja.md]: Yes working on this in separate PR now

**Comment by amiecorso** [comment]
yes! conflicts should be resolved and comments addressed

**Comment by amiecorso** [comment]
Just FYI Hans this is currently failing circleCI based on unrelated PayID Client tests

**Comment by keefertaylor** [comment]
Updating this branch since the test fixes landed. Sorry!

**Comment by keefertaylor** [comment]
> : the XRPClient integration tests were using an outdated transaction hash that had probably been wiped off testnet. Pulled the newer hash from the XpringClient integration tests and now it's good.

Great deduction. We should maybe / probably generate this hash on the fly so that we don't need to worry about this happening (I think the TestNet node we use only keeps ~1 month of history) so this is a 1x a month problem.

File: .mocharc.js (hunk 1 above)
Comment (keefertaylor) [reply .mocharc.js]: Oh, I think that when you autosave the file it formatted it with eslint. I had the same thing happen to me.

File: test/mocha.opts [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+--recursive
+test/**/*.ts
```
Comment (hbergren) [inline test/mocha.opts]: You said this was autogenerated, right?

File: test/xrp-client-integration-test.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+import { assert } from 'chai'
+import bigInt from 'big-integer'
+import XRPClient from '../src/xrp-client'
+import TransactionStatus from '../src/transaction-status'
+
+// A timeout for these tests.
+const timeoutMs = 60 * 1000 // 1 minute
+
+// An address on TestNet that has a balance.
+const recipientAddress = 'X7cBcY4bdTTzk3LHmrKAK6GyrirkXfLHGFxzke5zTmYMfw4'
```
Comment (hbergren) [inline test/xrp-client-integration-test.ts]: don't testnet addresses start with `T`?

File: test/mocha.opts (hunk 5 above)
Comment (keefertaylor) [reply test/mocha.opts]: We should remove this file. This used to be the way to configure mocha but now it's deprecated in favor of a `.mocharc.js` in the root of the repository, which we have.

File: test/xrp-client-integration-test.ts (hunk 6 above)
Comment (keefertaylor) [reply test/xrp-client-integration-test.ts]: Yeah, technically. In reality, the prefix is just meta data that the client side SDKs attach. The actual rippled APIs only take in classic address / destination tag so everything is decoded before it's sent to the server anyway. 

We can change this to: T7X5peX2VRppBeaupFh5D6p1VHYZpgzKDzvThKxXh4rTBfq by using https://xrpaddress.info/ if we wanted.

**Pull request merged**